packages/catalog/src/types.ts

```diff
@@ -22,6 +22,8 @@ export interface CatalogResultsProps {
   companyDefaultLocale?: string;
   /** currency code to format price */
   currencyCode?: string;
+  /** number of items to return */
+  numberOfContentItems?: number;
```

Comment:
Since this prop only exists in your custom component and not in the original catalog package, you'll want to extend the `CatalogResultsProps` (or the `CatalogProps`) interface in your custom component.